src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -0,0 +1,367 @@
+                        } else {
+                            this.nextStep();
+                        }
+                    });
+                } else {
+                    if (this._checkSelectorValidity()) {
+                        this._guidedTourCurrentStepSubject.next(this.getPreparedTourStep(this._currentTourStepIndex));
+                    } else {
+                        this.nextStep();
+                    }
+                }
+            } else {
+                if (this._currentTour.completeCallback) {
+                    this._currentTour.completeCallback();
+                }
+                this.updateGuidedTourSettings(this._currentTour.settingsId, false).subscribe(guidedTourSettings => {
+                    if (guidedTourSettings.body) {
+                        this.guidedTourSettings = guidedTourSettings.body;
+                    }
+                });
+                this.resetTour();
+            }
+        }
+    }
+
+    /**
+     * Navigate to previous tour step
+     */
+    public backStep(): void {
+        if (this._currentTour) {
+            const currentStep = this._currentTour.steps[this._currentTourStepIndex];
+            if (currentStep.closeAction) {
+                currentStep.closeAction();
+            }
+            if (this._currentTour.steps[this._currentTourStepIndex - 1]) {
+                this._currentTourStepIndex--;
+                this._setFirstAndLast();
+                if (currentStep.action) {
+                    currentStep.action();
+                    setTimeout(() => {
+                        if (this._checkSelectorValidity()) {
+                            this._guidedTourCurrentStepSubject.next(this.getPreparedTourStep(this._currentTourStepIndex));
+                        } else {
+                            this.backStep();
+                        }
+                    });
+                } else {
+                    if (this._checkSelectorValidity()) {
+                        this._guidedTourCurrentStepSubject.next(this.getPreparedTourStep(this._currentTourStepIndex));
+                    } else {
+                        this.backStep();
+                    }
+                }
+            } else {
+                this.resetTour();
+            }
+        }
+    }
+
+    /**
+     * Skip tour
+     */
+    public skipTour(): void {
+        if (this._currentTour) {
+            if (this._currentTour.skipCallback) {
+                this._currentTour.skipCallback(this._currentTourStepIndex);
+            }
+            this.resetTour();
+        }
+    }
+
+    /**
+     * Close tour and remove overlay
+     */
+    public resetTour(): void {
+        document.body.classList.remove('tour-open');
+        this._currentTour = undefined;
+        this._currentTourStepIndex = 0;
+        this._guidedTourCurrentStepSubject.next(undefined);
```

Comment:
It would be better imo to send null instead of undefined.
Undefined is not checked here by the TS compiler, if the type of the subject is `step | null`, subscribers will know that a null value can happen.